pkg/prom/collector.go

```diff
@@ -1323,6 +1326,51 @@ func (c *SolidfireCollector) collectVolumeAccessGroups(ctx context.Context, ch c
 	return nil
 }
 
+func (c *SolidfireCollector) collectVirtualVolumeTasks(ctx context.Context, ch chan<- prometheus.Metric) error {
+	vvt, err := c.client.ListVirtualVolumeTasks(ctx)
+	if err != nil {
+		return err
+	}
+	mu.Lock()
+	defer mu.Unlock()
+	ch <- prometheus.MustNewConstMetric(
+		MetricDescriptions.VirtualVolumeTasks,
+		prometheus.CounterValue,
+		float64(len(vvt.Result.Tasks)),
+	)
+	return nil
+}
+
+func (c *SolidfireCollector) collectBulkVolumeJobs(ctx context.Context, ch chan<- prometheus.Metric) error {
+	btj, err := c.client.ListBulkVolumeJobs(ctx)
+	if err != nil {
+		return err
+	}
+	mu.Lock()
+	defer mu.Unlock()
+	ch <- prometheus.MustNewConstMetric(
+		MetricDescriptions.BulkVolumeJobs,
+		prometheus.CounterValue,
+		float64(len(btj.Result.BulkVolumeJobs)),
+	)
+	return nil
+}
+
+func (c *SolidfireCollector) collectAsyncResults(ctx context.Context, ch chan<- prometheus.Metric) error {
```

Comment:
I think it'd be really helpful to split these into counts labelled by the type of async result (BulkVolume, Clone, etc)

Reply:
Done